File: randomNumber.py
import random
from exceptions import TooManyRequests, RequestTimeout, ServerDown


class RandomNumberGenerator:
    @staticmethod
    def generate_and_check():
        random_number = random.randint(1, 100)

        if random_number == 50:
            raise TooManyRequests()

        if random_number == 99:
            raise RequestTimeout()

        if random_number == 88:
            raise ServerDown()

File: src/main/java/co/hei/utilities/RandomNumber.java
package co.hei.utilities;

import co.hei.exceptions.RequestTimeOutException;
import co.hei.exceptions.ServerDownException;
import co.hei.exceptions.TooManyRequestsException;
import java.util.Random;

public class RandomNumber {
    public static void generateAndCheck() throws TooManyRequestsException, RequestTimeOutException, ServerDownException {
        Random random = new Random();
        int randomNumber = random.nextInt(100) + 1;

        if (randomNumber == 50) {
            throw new TooManyRequestsException("Too many requests in a short period of time");
        }

        if (randomNumber == 99) {
            throw new RequestTimeOutException("The request has expired");
        }

        if (randomNumber == 88) {
            throw new ServerDownException("The server is unavailable.");
        }
    }
}
